src/openrct2-ui/Ui.cpp

```diff
@@ -59,8 +59,12 @@ int main(int argc, const char** argv)
 
             context->RunOpenRCT2(argc, argv);
         }
+        return EXIT_SUCCESS;
+    }
+    else
+    {
+        return EXIT_FAILURE;
```

Comment:
I'm not sure if this is correct. There are many cases where OpenRCT2 simply prints something to the console and does not run the game - extracting sprites for example.

Reply:
I've spent a little more time with the game's code and have a couple of questions. Firstly, if it's common for fatal errors in the game to simply print a message, I wonder if issue #9477 is misguided in wanting to replace the use of gExitCode, I'm not sure how the solution proposed was intended to work across all relevant files. Perhaps it would make sense to employ gExitCode everywhere as it might have originally been intended. I would appreciate it if @ZehMatt could elaborate on the original intent of the issue.

My second question is concerning the sometimes already used exit(), is this only acceptable in some cases for potential issues with threads or would it be worth considering using it as the standard error termination?

To speak to what might be obvious, this is a very big and complex project for me, but I tend to learn and improve best when I throw myself off the deep end, so again, apologies for being new.

Reply:
We could do `std::exit` but that will not clean up any stack objects. When a fatal error occurs (which should really be rare anyway), the best thing to do is throw an exception which can then bubble up the program.

For this PR, `cmdline_run` can return 1 or 0 which just controls whether the main game should run or not. That function and its call chain would need to return a struct which contains exit code and whether it should exit. Or maybe just `std::optional` would work where `nullopt` states that it should run the main game.